practice_app/pages/api/getProductCatalog.js

```diff
@@ -0,0 +1,278 @@
+const getProductCatalog = (req, res) =>
+  new Promise(async (resolve) => {
+    const axios = require("axios");
+    // Imports the Google Cloud client library
+    const vision = require("@google-cloud/vision");
+    // Creates a client
+    const client = new vision.ProductSearchClient();
+    // Id of the getflix in google cloud storage
+    const projectId = "brilliant-era-276800";
+    // Location of the getflix in google cloud storage
+    const location = "europe-west1";
+    // Method of the request
+    const method = req.method;
+
+    switch (method) {
+      case "POST": {
+        const requestType = req.body.type;
+        // Creates a product and then adds it to the product set
+        if (requestType == "product") {
+          console.log("aaaaa");
+          console.log(req.body.setId);
+          // Gets body of request
+          const productId = req.body.id;
+          // Display name of the product
+          const productDisplayName = req.body.name;
+          // Category of the product
+          const productCategory = req.body.category;
+          // Id of the product set
+          const productSetId = req.body.setId;
+          // Description of the product
+          const productDesc = req.body.description;
+          // Color label of the product
+          const color = req.body.color;
+          // Style label of the product
+          const style = req.body.style;
+
+          // Resource path that represents Google Cloud Platform location.
+          const locationPath = client.locationPath(projectId, location);
+          // Initializes the product object
+          const product = {
+            displayName: productDisplayName,
+            productCategory: productCategory,
+            description: productDesc,
+            productLabels: [
+              {
+                key: "color",
+                value: color,
+              },
+              {
+                key: "style",
+                value: style,
+              },
+            ],
+          };
+
+          // Initializes the request object
+          const createRequest = {
+            parent: locationPath,
+            product: product,
+            productId: productId,
+          };
+          // Creates the product via 'createProduct' method of the Google Product Search API
+          await client.createProduct(createRequest).catch((err) => {
```

Comment:
I would suggest not mixing `async ... await` syntax with `.then(..).catch(...)`

https://css-tricks.com/understanding-async-await/